source/Test.cpp

```diff
@@ -97,6 +97,28 @@ namespace{
 		// SDL_PushEvent in the future to use the regular SDL event-handling loops.
 		return EventToUI(menuOrGamePanels, event);
 	}
+	
+	string ShipToString(const Ship &ship)
+	{
+		string shipDescription = "";
+		const System *shipSystem = ship.GetSystem();
+		const Planet *shipPlanet = ship.GetPlanet();
+		if(shipSystem)
+			shipDescription += "system: " + shipSystem->Name();
+		else
+			shipDescription += " system: <not set>";
+		if(shipPlanet)
+			shipDescription += ", planet:" + shipPlanet->TrueName();
+		else
+			shipDescription += ", planet: <not set>";
```

Comment:
You can use the conditional operator here:

```c++
description += ", planet: " + (planet ? planet->TrueName() : "<not set>"));
```

or split it:

```c++
description += ", planet: ";
description += planet ? planet->TrueName() : "<not set>";
```

Second snippet looks more readable imo

Reply:
I took the first snippet, it seemed to match the other lines better.